content/blog/duckplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,284 @@
+Not only is the syntax the same, the semantics are too!
+
+The duckplyr package is fully compatible with dplyr: if an operation cannot be carried out with DuckDB, it is automatically outsourced to dplyr.
+Over time, we expect fewer and fewer fallbacks to dplyr to be needed.
+
+
+## How to use duckplyr
+
+To _replace_ dplyr with duckplyr, you can:
+
+- Load duckplyr and then keep your pipeline as is. Calling `library(duckplyr)` overwrites dplyr methods, enabling duckplyr for the entire session no matter how data.frames are created.
+    This is shown in the example above.
+
+- Create individual "duck frames" using _conversion functions_ like `duckdb_tibble()` or `as_duckdb_tibble()`, or _ingestion functions_ like `read_csv_duckdb()`.
+
+
+Then, the data manipulation pipeline uses the exact same syntax as a dplyr pipeline.
+The duckplyr package performs the computation using DuckDB.
+
+```{r, echo=FALSE, message=FALSE, warning=FALSE}
+# Undo the effect of library(duckplyr)
+methods_restore()
+```
+
+
+```{r}
+out <- lineitem |>
+  duckplyr::as_duckdb_tibble() |> # this is the only change :-)
+  select(l_shipdate, l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  filter(l_shipdate <= !!as.Date("1998-09-02")) |>
+  select(l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  summarise(
+    sum_qty = sum(l_quantity),
+    sum_base_price = sum(l_extendedprice),
+    sum_disc_price = sum(l_extendedprice * (1 - l_discount)),
+    sum_charge = sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)),
+    avg_qty = mean(l_quantity),
+    avg_price = mean(l_extendedprice),
+    avg_disc = mean(l_discount),
+    count_order = n(),
+    .by = c(l_returnflag, l_linestatus)
+  ) |>
+  arrange(l_returnflag, l_linestatus)
+```
+
+The result can finally be materialized to memory, or computed temporarily, or computed to a file.
+
+```{r}
+# to memory
+nrow(out)
+# or for instance collect(out)
+
+# to a file
+csv_file <- withr::local_tempfile()
+compute_csv(out, csv_file)
+fs::file_size(csv_file)
+```
+
+Operations not yet supported by duckplyr are automatically outsourced to dplyr.
+For instance, filtering on grouped data is not supported yet, still it works thanks to the fallback mechanism.
+
+```{r}
+# filter(n == max(n), .by = c(year, sex)
+lineitem |>
+  duckplyr::as_duckdb_tibble() |>
+  select(l_shipdate, l_returnflag, l_linestatus, l_quantity, l_extendedprice, l_discount, l_tax) |>
+  filter(l_quantity == max(l_quantity), .by = c(l_returnflag, l_linestatus))
+```
+
+For performance reasons, the output order of the result is not guaranteed to be stable.
+If you need a stable order, you can use `arrange()` or force output order stability by setting an environment variable.
+This and other limitations are documented in [`vignette("limits")`](https://duckplyr.tidyverse.org/articles/limits.html).
+
```

Comment:
This now has little value because our TPC-H example has a final `arrange()`, and nobody would be able to tell from our new example unless we show them.

```suggestion
```